kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -792,90 +810,97 @@ class ResidualBasedBlockBuilderAndSolver
      * should refer to the particular Builder And Solver choosen
      * @param pScheme The integration scheme considered
      * @param rModelPart The model part of the problem to solve
-     * @param A The LHS matrix
-     * @param Dx The Unknowns vector
-     * @param b The RHS vector
+     * @param rA The LHS matrix
+     * @param rDx The Unknowns vector
+     * @param rb The RHS vector
      */
     void ApplyDirichletConditions(
         typename TSchemeType::Pointer pScheme,
         ModelPart& rModelPart,
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b) override
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
     {
-        std::size_t system_size = A.size1();
-        std::vector<double> scaling_factors (system_size, 0.0);
+        std::size_t system_size = rA.size1();
+        std::vector<double> scaling_factors (system_size, 1.0);
 
+        const auto it_dof_iterator_begin = BaseType::mDofSet.begin();
         const int ndofs = static_cast<int>(BaseType::mDofSet.size());
 
-        //NOTE: dofs are assumed to be numbered consecutively in the BlockBuilderAndSolver
-        #pragma omp parallel for firstprivate(ndofs)
+        // NOTE: dofs are assumed to be numbered consecutively in the BlockBuilderAndSolver
+        #pragma omp parallel for
         for (int k = 0; k<ndofs; k++) {
-            typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin() + k;
-            if(dof_iterator->IsFixed())
+            auto it_dof_iterator = it_dof_iterator_begin + k;
+            if (it_dof_iterator->IsFixed())

```

Comment:
actually if you were using a Vector (admittedly not for an std::vector) it would be better to leave the previous version, as this way initialization happens in parallel and the "first touching" is performed correctly

Reply:
as i am telling here using a vector is better, and the old code is better than the new one